### PR TITLE
network: fix an infinite loop

### LINKED ISSUE #19467: network: segfault when the self interface is specified in Bridge=
**systemd version the issue has been seen with**
247.3-5

**Used distribution**
Debian GNU/Linux

**Linux kernel version used** (`uname -a`)

 `Linux localhost 5.10.0-6-amd64 #1 SMP Debian 5.10.28-1 (2021-04-09) x86_64 GNU/Linux`

**CPU architecture issue was seen on**
amd64

**Expected behaviour you didn't see**
A functional network configuration as configured

**Unexpected behaviour you saw**
a) the system is offline
b) systemd-networkd.service crashes with a segmentation fault
c) systemd-netword.service tells me I'd try to "enslave a bridge to a bridge" (Hum, where do I try to "enslave a bridge to a bridge"?)

**Steps to reproduce the problem**
1) add `dhcp=1` to your Kernel command line 
2) create some .netdev, .network and .conf files

2.1) `/etc/systemd/network/br0.netdev`:
  > [NetDev]
  > Name=br0
  > Kind=bridge
  > MACAddress=A9:97:98:11:43:79

  2.2) `/etc/systemd/network/default_dhcp.network`:
  > [Match]
  > KernelCommandLine=dhcp
  >
  > [Network]

  2.3) `/etc/systemd/network/default_dhcp.network.d/br0_dhcp.network.conf`:
  > [Match]
  > Name=br0
  > KernelCommandLine=dhcp
  >
  > [Network]
  > DHCP=ipv4
  >
  > [DHCPv4]
   > Hostname=d15
  > UseDomains=true

2.4) `/etc/systemd/network/default_dhcp.network.d/bridged_enp11s0.network.conf`
  > [Match]
  > Name=enp11s0
  > KernelCommandLine=dhcp
  >
  > [Network]
  > Bridge=br0

**Additional program output to the terminal or log subsystem illustrating the issue**

**`journalctl -u systemd-networkd`**
```
--  Boot 31deae5599ad47df880be26210935b4a --
Apr 29 11:38:28 d15 systemd-networkd[309]: br0: netdev ready
Apr 29 11:38:28 d15 systemd-networkd[309]: Enumeration completed
Apr 29 11:38:28 d15 systemd-networkd[309]: br0: Could not join netdev: Can not enslave a bridge to a bridge. Too many levels of symbolic links
Apr 29 11:38:28 d15 systemd-networkd[309]: br0: Failed
Apr 29 11:38:28 d15 systemd[1]: Started Network Service.
Apr 29 11:38:29 d15 systemd[1]: systemd-networkd.service: Main process exited, code=dumped, status=11/SEGV
Apr 29 11:38:29 d15 systemd[1]: systemd-networkd.service: Failed with result 'core-dump'.
Apr 29 11:38:29 d15 systemd[1]: systemd-networkd.service: Scheduled restart job, restart counter is at 1.
Apr 29 11:38:29 d15 systemd[1]: Stopped Network Service.
Apr 29 11:38:29 d15 systemd[1]: Starting Network Service...
Apr 29 11:38:29 d15 systemd-networkd[530]: br0: netdev ready
Apr 29 11:38:29 d15 systemd-networkd[530]: Enumeration completed
Apr 29 11:38:29 d15 systemd[1]: Started Network Service.
Apr 29 11:38:29 d15 systemd-networkd[530]: br0: netdev exists, using existing without changing its parameters
Apr 29 11:38:29 d15 systemd-networkd[530]: br0: Could not join netdev: Can not enslave a bridge to a bridge. Too many levels of symbolic links
Apr 29 11:38:29 d15 systemd-networkd[530]: br0: Failed
Apr 29 11:38:29 d15 systemd[1]: systemd-networkd.service: Main process exited, code=dumped, status=11/SEGV
Apr 29 11:38:29 d15 systemd[1]: systemd-networkd.service: Failed with result 'core-dump'.
Apr 29 11:38:29 d15 systemd[1]: systemd-networkd.service: Scheduled restart job, restart counter is at 2.
Apr 29 11:38:29 d15 systemd[1]: Stopped Network Service.
Apr 29 11:38:29 d15 systemd[1]: Starting Network Service...
Apr 29 11:38:29 d15 systemd-networkd[568]: br0: netdev ready
Apr 29 11:38:29 d15 systemd-networkd[568]: Enumeration completed
Apr 29 11:38:29 d15 systemd[1]: Started Network Service.
Apr 29 11:38:29 d15 systemd-networkd[568]: br0: netdev exists, using existing without changing its parameters
Apr 29 11:38:29 d15 systemd-networkd[568]: br0: Could not join netdev: Can not enslave a bridge to a bridge. Too many levels of symbolic links
Apr 29 11:38:29 d15 systemd-networkd[568]: br0: Failed
Apr 29 11:38:32 d15 systemd[1]: systemd-networkd.service: Main process exited, code=dumped, status=11/SEGV
Apr 29 11:38:32 d15 systemd[1]: systemd-networkd.service: Failed with result 'core-dump'.
Apr 29 11:38:32 d15 systemd[1]: systemd-networkd.service: Scheduled restart job, restart counter is at 3.
Apr 29 11:38:32 d15 systemd[1]: Stopped Network Service.
Apr 29 11:38:32 d15 systemd[1]: Starting Network Service...
Apr 29 11:38:33 d15 systemd-networkd[657]: br0: netdev ready
Apr 29 11:38:33 d15 systemd-networkd[657]: Enumeration completed
Apr 29 11:38:33 d15 systemd[1]: Started Network Service.
Apr 29 11:38:33 d15 systemd-networkd[657]: br0: netdev exists, using existing without changing its parameters
Apr 29 11:38:33 d15 systemd-networkd[657]: br0: Could not join netdev: Can not enslave a bridge to a bridge. Too many levels of symbolic links
Apr 29 11:38:33 d15 systemd-networkd[657]: br0: Failed
Apr 29 11:42:03 d15 systemd[1]: Stopping Network Service...
Apr 29 11:42:03 d15 systemd[1]: systemd-networkd.service: Succeeded.
Apr 29 11:42:03 d15 systemd[1]: Stopped Network Service.
Apr 29 11:42:03 d15 systemd[1]: Starting Network Service...
Apr 29 11:42:03 d15 systemd-networkd[828]: br0: netdev ready
Apr 29 11:42:03 d15 systemd-networkd[828]: Enumeration completed
Apr 29 11:42:03 d15 systemd[1]: Started Network Service.
Apr 29 11:42:03 d15 systemd-networkd[828]: br0: netdev exists, using existing without changing its parameters
Apr 29 11:42:04 d15 systemd-networkd[828]: br0: Could not join netdev: Can not enslave a bridge to a bridge. Too many levels of symbolic links
Apr 29 11:42:04 d15 systemd-networkd[828]: br0: Failed
Apr 29 11:44:19 d15 systemd[1]: Stopping Network Service...
Apr 29 11:44:19 d15 systemd[1]: systemd-networkd.service: Succeeded.
Apr 29 11:44:19 d15 systemd[1]: Stopped Network Service.
```
And here's some `coredumpctl debug $ID` output:

[coredumpctl_debug_309.txt](https://github.com/systemd/systemd/files/6399101/coredumpctl_debug_309.txt)
[coredumpctl_debug_530.txt](https://github.com/systemd/systemd/files/6399102/coredumpctl_debug_530.txt)
[coredumpctl_debug_568.txt](https://github.com/systemd/systemd/files/6399103/coredumpctl_debug_568.txt)

If you need more details, please let me know wich gdb-commands I have to use.

### PR DESCRIPTION
**Pull request by yuwata**
Fixes #19467.

### CODE REVIEW

File: src/network/networkd-link.c [hunk 1]
```diff
@@ -179,6 +179,12 @@ static void link_update_master_operstate(Link *link, NetDev *netdev) {
         if (netdev->ifindex <= 0)
                 return;
 
+        /* If the interface is self-mentioned in Bridge= or friends, then this introdduce infinite loop.
```
Comment (keszybz) [inline src/network/networkd-link.c]: Typo here. But since this will need to be reworked for the FIXME, let's fix that then.

**Review by keszybz — COMMENTED** [review]
LGTM.

**Comment by keszybz** [comment]
CI failed because of the GREEDY_REALLOC issue. I restart some of the tests.

**Comment by yuwata** [comment]
Rebased, and typos are fixed.

**Pull request merged**